File: cw/python/break_camelcase.py
# https://www.codewars.com/kata/5208f99aee097e6552000148/train/python


import re

def solution(s):
    camel = re.findall(r'^[a-z$]+', s)
    case = re.findall(r'[A-Z][^A-Z]+', s)

    return ' '.join([i for i in (camel + case)])


if __name__ == '__main__':
    s = input()
    print(solution(s))

File: codewars/rust/src/break_camelcase.rs
// https://www.codewars.com/kata/5208f99aee097e6552000148/train/rust

use std::io;
use regex::Regex;

fn solution(s: &str) -> String {
	let camel = Regex::new(r"^[a-z$]+").unwrap()
		.find(s).unwrap()
		.as_str();

	let mut case = String::new();

	for i in Regex::new(r"[A-Z][a-z]+").unwrap().find_iter(s) {
		case.push_str(&format!("{} ", i.as_str()));
	}

	format!("{} {}", &camel, &case.trim()).to_string()
}

fn main() {
	let mut s = String::new();

	io::stdin()
		.read_line(&mut s)
		.expect("Failed to read line");

	println!("{}", solution(&s));
}